### PR TITLE
use list::swap for better performance

### PR DESCRIPTION
**Pull request by cfanz**
use swap function to avoid extra copy constructions.

### CODE REVIEW

File: src/common/OutputDataSocket.cc [hunk 1]
```diff
@@ -299,8 +299,7 @@ int OutputDataSocket::dump_data(int fd)
 {
   m_lock.Lock(); 
   list<bufferlist> l;
-  l = data;
-  data.clear();
+  l.swap(data);
```
Comment (tchaikov) [inline src/common/OutputDataSocket.cc]: use `std::move()` for better performance.

**Comment by tchaikov** [comment]
could you prefix the title of your commit message with the subcomponent your are changing ? see https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#3-describe-your-changes

also, 
could you add a "`Signed-off-by`" line at the end of your commit message? "git commit -s" will do the trick for you. see https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#1-sign-your-work

File: src/common/OutputDataSocket.cc (hunk 1 above)
Comment (joscollin) [inline src/common/OutputDataSocket.cc]: @tchaikov That's a move operation. But I've been thinking, is push_back() a valid operation on a moved from object (data) ?

Comment (tchaikov) [inline src/common/OutputDataSocket.cc]: @joscollin you can try to implement `swap()`, and see what the difference it is.

Comment (joscollin) [inline src/common/OutputDataSocket.cc]: @tchaikov Sorry, if I confused you. I agree that the std::move is required. But I was thinking about the `data.push_back()` at L315.

Comment (cbodley) [inline src/common/OutputDataSocket.cc]: i'd expect this:
```c++
  list<bufferlist> l;
  l.swap(data);
```
to be equivalent to this:
```c++
  list<bufferlist> l = std::move(data);
  data.clear();
```

**Comment by cfanz** [comment]
@tchaikov Thanks for your advices. I've created a new pull request for this. See https://github.com/ceph/ceph/pull/16548.

**Pull request closed without merging**